### ISSUE TITLE
Header / Footer

### ISSUE THREAD

**Issue by jfairley** [Ziv-Barber/officegen#52]
It would be great to build headers and footers into a word document. It looks like the xml syntax is the same if not restricted, except there's "header1.xml" and "footer1.xml" in addition to "document.xml". Thoughts?

**Comment by Ziv-Barber** [comment]
I started to investigate it back then.

**Comment by jfairley** [comment]
I have a change in progress, though admittedly, I chose to omit the options for now.

I'm on the fence on one issue. In a footer there's typically something left-aligned _and_ something right-aligned (_and_ sometimes something center-aligned). I didn't see how to do that with the existing patterns you've implemented, so I was considering building in three functions -- one to take left-aligned text, one to take centered text, and one to take right-aligned text.

Maybe something like this.

```javascript
        var report = officegen({
            type: 'docx',
            title: 'Delivery Schedule'
        });

        report.getHeader().setTextLeft('header-left');
        report.getHeader().setTextCenter('header-center');
        report.getHeader().setTextRight('header-right');

        report.getFooter().setTextLeft('left');
        report.getFooter().setTextCenter('center');
        report.getFooter().setTextRight('right');
```

What are your thoughts on this?

**Comment by Ziv-Barber** [comment]
Yes, sounds like a good plan.

**Comment by jfairley** [comment]
Sorry for the delay. Just letting you know this is still on my task list. I had an issue with spacing, then got pulled away to another issue. Definitely gonna come back to finish this thought.

**Comment by nanthak** [comment]
i can't use headers in docx as jfairley said. This is my code snippet.

        docx.getHeader().setTextLeft('header-left');
        docx.getHeader().setTextCenter('header-center');
        docx.getHeader().setTextRight('header-right');
        docx.getFooter().setTextLeft('left');
        docx.getFooter().setTextCenter('center');
        docx.getFooter().setTextRight('right');

In my project i have to generate a report which may be in pdf or word. I have done with pdf. but not on word(docx). I need to done generating docx file that should contains,
      1.header with logo img on the top right corner.
      2. in the body of docx, have to generate a html table with external css. The content of the html is varied depends on the user selection from client side.
      3. Footer with page number.


Need your help to do this. Hope you will...

Thanks..

**Comment by nanthak** [comment]
Hi jfairley,

Can you please place your code here to know more about this. I'm interested in your way of building headers and footers. Please, do let me know once you completed this module. I'm looking out for your release.

**Comment by jfairley** [comment]
@Ziv-Barber. I said I would finish this, and I never got around to it. Sorry. I put what I have in PR: https://github.com/Ziv-Barber/officegen/pull/67. It definitely works, but my attempts to build in alignment gutters wasn't coming out right. (Everything comes out left-aligned for me.)

@nanthak, I hope it is helpful for you.

**Comment by nanthak** [comment]
Thank you jfairley. You have a done a great job. I have one more issue is that i can't create a table with this. even i have a addCell() method, my table won't be appeared properly. I will show my output in the following image.
![screenshot from 2015-07-01 13 10 25](https://cloud.githubusercontent.com/assets/13063818/8449849/945a0dd8-1ff2-11e5-8e2d-6328de6b5c59.png)

**Comment by jchabot86** [comment]
so is this feature available if i install from npm?

**Comment by Ziv-Barber** [comment]
Header and footer coming soon. I already implemented most of it.

**Comment by Ziv-Barber** [comment]
Implemented for version 0.4.2 (only in Github, soon also for npm after finishing to implement more features).
As for page numbers (and more field objects), coming soon.
